include/tvm/topi/einsum.h

```diff
@@ -49,623 +49,15 @@ namespace topi {
-        second.append(1, c);
-      }
-    }
-    ret = std::make_tuple(first, second);
-  }
-
-  // Make sure output subscripts are in the input
-  std::bitset<LABELRANGE> input_subscripts_set = Str2Set(std::get<0>(ret));
-  for (const char& c : std::get<1>(ret)) {
-    CHECK(input_subscripts_set.test(c))
-        << "Output character " << c << " did not appear in the input";
-  }
-
-  // Make sure number operands is equivalent to the number of terms
-  CHECK_EQ(std::count(std::get<0>(ret).begin(), std::get<0>(ret).end(), ',') + 1, operands.size())
-      << "Number of einsum subscripts must be equal to the "
-      << "number of operands.";
-
-  return ret;
-}
-
 /*!
  * \brief Compute the shape of the output.
  * \param subscripts input subscripts.
  * \param operands operand tensors.
  *
  * \return the shape of the output.
  */
-inline Array<PrimExpr> NumpyEinsumShape(const std::string subscripts,
-                                        const std::vector<Array<PrimExpr>>& operands) {
-  // Parsing
-  std::tuple<std::string, std::string> parsed_subscripts = ParseEinsumInput(subscripts, operands);
-
-  // Build a few useful list and sets
-  std::vector<std::string> input_list = Split(std::get<0>(parsed_subscripts), ",");
-  size_t isize = input_list.size();
-
-  // Get length of each unique dimension and ensure all dimensions are correct
-  int dimension_dict[LABELRANGE];
-  memset(dimension_dict, -1, sizeof(dimension_dict));
-  for (size_t i = 0; i < isize; ++i) {
-    const std::string& term = input_list[i];
-    const Array<PrimExpr>& sh = operands[i];
-    CHECK_EQ(sh.size(), term.length())
-        << "Einstein sum subscript " << input_list[i] << " does not contain the "
-        << "correct number of indices for operand " << i << ".";
-    size_t len_term = term.length();
-    for (size_t j = 0; j < len_term; ++j) {
-      int64_t dim = GetConstInt(sh[j]);
-      const char& c = term[j];
-
-      if (dimension_dict[static_cast<int>(c)] != -1) {
-        // For broadcasting cases we always want the largest dim size
-        if (dimension_dict[static_cast<int>(c)] == 1) {
-          dimension_dict[static_cast<int>(c)] = dim;
-        }
-        CHECK(dim == 1 || dim == dimension_dict[static_cast<int>(c)])
-            << "Size of label '" << c << "' for operand  " << i << " ("
-            << dimension_dict[static_cast<int>(c)] << ") does not match previous terms (" << dim
-            << ").";
-      } else {
-        dimension_dict[static_cast<int>(c)] = dim;
-      }
-    }
-  }
-
-  // Get oshape
-  const std::string& output_str = std::get<1>(parsed_subscripts);
-  size_t odim = output_str.size();
-  Array<PrimExpr> oshape(odim, -1);
-  for (size_t i = 0; i < odim; ++i) {
-    oshape.Set(i, dimension_dict[static_cast<int>(output_str[i])]);
-  }
-  // Neglecting oshape assign check temporally
-  return oshape;
-}
+Array<PrimExpr> NumpyEinsumShape(const std::string subscripts,
```

Comment:
Just curious: what does the “Numpy” prefix stand for?

Reply:
this is the name used previously, indeed there's no reason using numpy prefix, we can probably find a better name

Reply:
Sounds good. The function is just doing pure shape inference... Just some random thought, what about “InferEinsumShape”?